src/luascript.cpp

```diff
@@ -17746,8 +17739,8 @@ int LuaScriptInterface::luaWeaponRegister(lua_State* L)
 		}
 
 		weapon->configureWeapon(it);
-		tfs::lua::pushBoolean(L, g_weapons->registerLuaEvent(weapon));
-		*weaponPtr = nullptr; // Remove luascript reference
+		*weaponPtr = nullptr; // Remove luascript reference before transferring ownership
+		tfs::lua::pushBoolean(L, g_weapons.registerLuaEvent(Weapon_ptr(weapon)));
```

Comment:
You can push weapons to Lua as shared ptrs now, using `pushSharedPtr` and `getSharedPtr` and avoid all manual memory management